thirdparty/faiss/faiss/utils/distances.h

```diff
@@ -234,7 +236,7 @@ void all_inner_product(
         size_t d,
         size_t nx,
         size_t ny,
-        std::vector<std::pair<float, int64_t>>& output,
+        std::vector<knowhere::DistId>& output,
```

Comment:
Is it possible to avoid modifying Faiss in this case?

Reply:
It adds more pain to me whenever I need to propagate the changes from the Faiss baseline. So, I expect and insist that the changes to the faiss core code to be the last resort.
But it should be acceptable if the functions are not from the original faiss.
Could you please check whether you'll be able to put your change on top of #453 and see whether they are compatible? Just check, don't rebase yet :) Thanks

Reply:
I just tried rebasing a copy of `iterator-new-interface` onto `merge_faiss_changes_from_baseline_v2a` and yes they are compatible. The only conflict is `CollectAllResultHandler` which is easy to fix.